python/ray/tune/trial_executor.py

```diff
@@ -177,7 +187,19 @@ def force_reconcilation_on_next_step_end(self):
     def on_no_available_trials(self, trial_runner):
         if self._queue_trials:
             return
-        for trial in trial_runner.get_trials():
+
+        all_trials = trial_runner.get_trials()
+        all_trials_are_pending = all(
+            trial.status == Trial.PENDING for trial in all_trials)
+        if all_trials_are_pending and not is_ray_cluster() and (
+                trial_runner.iteration +
+                1) % SHOW_MAYBE_INSUFFICIENT_RESOURCE_WARNING_ITER_DELAY == 0:
+            logger.warning(
```

Comment:
I think we should be using `warnings.warn` here instead, according to https://docs.python.org/3/howto/logging.html. We are not consistent with it in existing code but it would be good to use best practices in new code

Reply:
Those are good points. We should keep it as logging, then :+1: